### ISSUE TITLE
Proposal: new transaction boundary in 2.2

### ISSUE THREAD

**Issue by seratch** [scalikejdbc/scalikejdbc#302]
Currently, `DBConnection#localTx { session => }` (`DB.localTx` is also same) rolls back its transaction only when exception is thrown toward outside of the block. 

@gakuzzzz's PR https://github.com/scalikejdbc/scalikejdbc/pull/300 introduced `DBConnection#generalizedLocalTx[A]` and `TxBoundary[A]` type class. `TxBoundary` enables to customize the way to handle transactions for result types. That was so great work!

Since we already have `#futureLocalTx[A]`, I thought we should have `#tryLocalTx[A]` and `#eitherLocalTx[A]` too for API consistency. This PR today. https://github.com/scalikejdbc/scalikejdbc/pull/301

On the other hand, I probably should consider changing `DBConnection#localTx`'s behavior because `Try[A]`, `Either[Throwable, A]` and `Future[A]` have their own success/failure state which can be transaction boundary.

For instance, is there anyone who expects the transaction committed when the `Try` is a failure? I think no one expects such a behavior. 

``` scala
val member: Try[Member] = DB.localTx { implicit s =>
  Try {
     val groupId = Group.create("ScalikeJDBC User Group")
     val id = Member.create("Kaz", groupId)
     Member.findById(id)
  }
}
```

We can say the same for `Either[Throwable, A]` (not `Either[B, A]`).

Of course, existing application code may be accordingly working as expected by depending current ScalikeJDBC's hebavior. So we must avoid introducing this breaking change within version 2.1.x.

In conclusion, I'm thinking the following plan in 2.2. If you have any feedback, please tell us.
- Change `#localTx` to work as same as `#generalizedLocalTx` (perform rollback when exception is thrown or results represent failure due to exception - result types: Try[A], Either[Throwable, A] and Future[A] by default)
- Mark using `#generalizedLocalTx` directly as deprecated (and it will be removed since 2.3 or later)
- Mark `#futureLocalTx` as deprecated (and it will be removed since 2.3 or later)
